pilot/pkg/serviceregistry/external/servicediscovery.go

```diff
@@ -225,11 +235,81 @@ func getServiceEntryHandler(c *ServiceEntryStore) func(model.Config, model.Confi
 	}
 }
 
+// GetForeignServiceInstanceHandler defines the handler for service instances generated by other registries
+func (d *ServiceEntryStore) GetForeignServiceInstanceHandler() func(*model.ServiceInstance, model.Event) {
+	return func(si *model.ServiceInstance, event model.Event) {
+		key := configKey{
+			kind:      foreignInstanceConfigType,
+			name:      si.Endpoint.Address,
+			namespace: si.Service.Attributes.Namespace,
+		}
+		// Used to indicate if this event was fired for a pod->workloadentry conversion
+		// and that the event can be ignored due to no relevant change in the workloadentry
+		redundantEventForPod := false
+
+		d.storeMutex.Lock()
+		// We will only select entries in the same namespace
+		entries := d.seWithSelectorByNamespace[si.Service.Attributes.Namespace]
+
+		// this is from a pod. Store it in separate map so that
```

Comment:
How do you know it's from a pod ?

Reply:
We dont but we know its some service instance. I can change th etext